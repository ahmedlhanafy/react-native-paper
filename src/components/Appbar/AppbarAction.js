/* @flow */

import * as React from 'react';
import IconButton from '../IconButton';
import type { IconSource } from '../Icon';
import type { Theme } from '../../types';
import { withTheme } from '../../core/theming';

type Props = React.ElementConfig<typeof IconButton> & {|
  /**
   *  Custom color for action icon.
   */
  color?: string,
  /**
   * Name of the icon to show.
   */
  icon: IconSource,
  /**
   * Optional icon size.
   */
  size?: number,
  /**
   * Whether the button is disabled. A disabled button is greyed out and `onPress` is not called on touch.
   */
  disabled?: boolean,
  /**
   * Accessibility label for the button. This is read by the screen reader when the user taps the button.
   */
  accessibilityLabel?: string,
  /**
   * Function to execute on press.
   */
  onPress?: () => mixed,
  style?: any,
  /**
   * @optional
   */
  theme: Theme,
|};

/**
 * A component used to display an action item in the appbar.
 */
class AppbarAction extends React.Component<Props> {
  static displayName = 'Appbar.Action';

  static defaultProps = {
    size: 24,
  };

  render() {
    const {
      color: iconColor,
      icon,
      disabled,
      onPress,
      accessibilityLabel,
      theme: { colors },
      ...rest
    } = this.props;

    return (
      <IconButton
        onPress={onPress}
        color={iconColor || colors.typography.secondary}
        icon={icon}
        disabled={disabled}
        accessibilityLabel={accessibilityLabel}
        animated
        {...rest}
      />
    );
  }
}

export default withTheme(AppbarAction);
