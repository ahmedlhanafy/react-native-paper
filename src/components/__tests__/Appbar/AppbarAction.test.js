import React from 'react';
import renderer from 'react-test-renderer';
import AppbarAction from '../../Appbar/AppbarAction';

describe('ApAppbarActionpbar', () => {
  it('defaults its colors to colors.typography.secondary', () => {
    const tree = renderer.create(<AppbarAction icon="add-a-photo" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('accepts a color as props', () => {
    const tree = renderer
      .create(<AppbarAction icon="add-a-photo" color="#fff" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
