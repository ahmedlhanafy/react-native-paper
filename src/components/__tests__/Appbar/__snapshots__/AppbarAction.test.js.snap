// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApAppbarActionpbar accepts a color as props 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  hitSlop={
    Object {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "margin": 6,
          "overflow": "visible",
          "width": 36,
        },
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "height": 24,
            "width": 24,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ApAppbarActionpbar defaults its colors to colors.typography.secondary 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  hitSlop={
    Object {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "margin": 6,
          "overflow": "visible",
          "width": 36,
        },
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "height": 24,
            "width": 24,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
